Handle DB errors in tags similar module

alikon opened the pull request:
updated to handle DB errors like #6612 ,#6632, #6665, #6666, #6675 and for #6591

Testing info
simulate a db error

alikon committed:
Handle DB errors in tags similar module

updated to handle DB errors like #6612 ,#6632, #6665, #6666, #6675 and for #6591

Testing info
simulate a db error

alikon committed:
travis fix

travis fix

alikon committed:
travis

tab

nonumber commented:
Not sure if this should get merged with regards to discussion on https://github.com/joomla/joomla-cms/pull/6710

alikon committed:
added the generic error msg

used the generic message string "There has been an error" as suggested in #6710 the string was added in #7073
